src/common/middlewares/debug-logger.js

```
@@ -0,0 +1,59 @@
+  { flags: 'a' }
+);
+
+// Debug logger for services
+export const debugLogger = (context) => {
+  return {
+    debug: (message, data) => {
+      if (process.env.DEBUG_MODE !== 'true') return;
+      const timestamp = new Date().toISOString();
+      const logMessage = `[${timestamp}] [DEBUG] [${context}]: ${message} ${data ? JSON.stringify(data) : ''}`;
+      console.debug("\x1b[34m%s\x1b[0m", logMessage);
+      debugLogStream.write(logMessage + '\n');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)